custom_components/mc66c/sensor.py

```diff
@@ -92,7 +92,11 @@ def validateAndSetData(self, reading):
             _LOGGER.info("Successfully fetched new data: %s", reading)
 
             try:
-                data.append(int((reading[0]).decode("utf-8")) / 1000) # Energy.
+                for c in reading:
+                    if (len(c)) != 7:
+  	                    raise Exception('Received invalid datafield length')
+
+                data.append(int((reading[0]).decode("utf-8")) / 100) # Energy.
```

Comment:
```suggestion
                data.append(int((reading[0]).decode("utf-8")) / 1000) # Energy.
```

Maybe it's better to keep this `/1000` for backward compatibility? I've no idea how many people are using this component, so I can't estimate the impact this might have...

I do like your idea of adding a `config_flow` for this. You can have a look at the PR I did for another component: https://github.com/golles/ha-kamstrup_403/pull/15/files
Good to do that in another PR